stdlib/public/core/CocoaArray.swift

```diff
@@ -56,7 +56,7 @@ internal struct _CocoaArrayWrapper: RandomAccessCollection {
 
   @usableFromInline
   internal subscript(i: Int) -> AnyObject {
-    return core.objectAt(i)
+    return core.objectAt(i).takeUnretainedValue()
```

Comment:
><

Reply:
Won't this still do a retain/autorelease?

Reply:
I need to figure out how important this is. I’m hoping it’s a lot less hot than the actual bridging paths.

Reply:
Ok, checked on the generated code, and it's bad, but it at least doesn't autorelease since it can return +1:


                     _$ss18_CocoaArrayWrapperVyyXlSicig:        // Swift._CocoaArrayWrapper.subscript.getter : (Swift.Int) -> Swift.AnyObject
000000000000fc70         pushq      %rbp
000000000000fc71         movq       %rsp, %rbp
000000000000fc74         pushq      %r15
000000000000fc76         pushq      %r14
000000000000fc78         pushq      %rbx
000000000000fc79         pushq      %rax
000000000000fc7a         movq       %rsi, %rbx
000000000000fc7d         movq       %rdi, %r14
000000000000fc80         movq       0x3ad5f0, %r15                              ; @selector(objectAtIndex:)
000000000000fc87         movq       %rsi, %rdi                                  ; argument #1 for method _swift_unknownObjectRetain
000000000000fc8a         call       _swift_unknownObjectRetain                  ; _swift_unknownObjectRetain
000000000000fc8f         movq       %rbx, %rdi                                  ; argument "instance" for method imp___stubs__objc_msgSend
000000000000fc92         movq       %r15, %rsi                                  ; argument "selector" for method imp___stubs__objc_msgSend
000000000000fc95         movq       %r14, %rdx
000000000000fc98         call       imp___stubs__objc_msgSend                   ; objc_msgSend
000000000000fc9d         movq       %rax, %r14
000000000000fca0         movq       %rax, %rdi                                  ; argument #1 for method _swift_unknownObjectRetain
000000000000fca3         call       _swift_unknownObjectRetain                  ; _swift_unknownObjectRetain
000000000000fca8         movq       %rbx, %rdi                                  ; argument #1 for method _swift_unknownObjectRelease
000000000000fcab         call       _swift_unknownObjectRelease                 ; _swift_unknownObjectRelease
000000000000fcb0         movq       %r14, %rax
000000000000fcb3         addq       $0x8, %rsp
000000000000fcb7         popq       %rbx
000000000000fcb8         popq       %r14
000000000000fcba         popq       %r15
000000000000fcbc         popq       %rbp
000000000000fcbd         ret

Reply:
Sure, but…not sure you *want* to return at +1 if you're using this to implement `objectAtIndex:`.